patches/chrome-browser-ui-views-toolbar-toolbar_view.cc.patch

```diff
@@ -0,0 +1,72 @@
+diff --git a/chrome/browser/ui/views/toolbar/toolbar_view.cc b/chrome/browser/ui/views/toolbar/toolbar_view.cc
+index 3965bc59d1c67af0b84baa052054a5f854904533..ba6f8b6102fb6171d58739eddcda1532af8e6e25 100644
+--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
++++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
+@@ -69,6 +69,7 @@
+ #include "ui/views/widget/tooltip_manager.h"
+ #include "ui/views/widget/widget.h"
+ #include "ui/views/window/non_client_view.h"
++#include "brave/browser/ui/views/toolbar/bookmark_button.h"
```

Comment:
This patching is very bad in this file but I don't have a good idea on how to improve it.

Reply:
I think most likely by subclassing

Reply:
but actually in this case I think we want to just replace LocationBarView. We'll remove the original StarView using the NullView you suggested and then create a new view that adds the LocationBarView and the StarView/Bookmark Button as child views

Reply:
@bridiver that's an interesting idea. Some concerns before I proceed with it that I'd appreciate your thoughts on:

- `location_bar` is a public accessibly member of `ToolbarView`. Is there an easy way to make our new `BookmarkButtonAndLocationBarView` pass through all direct calls to public `LocationBarView` members to its own actual `location_bar` member?

- When layout out the view in the x axis, chromium uses the value `TOOLBAR_ELEMENT_PADDING` for the gap between each icon, but the value `TOOLBAR_STANDARD_SPACING` for the gap between the last icon (home, reload, or bookmark in Brave's case) and the location bar. If we inserted the icon in to the left of the brave location bar, then it would have incorrect spacing. If `TOOLBAR_STANDARD_SPACING` is less than `TOOLBAR_ELEMENT_PADDING` then we could add in the difference, but it feel like a hack.

- This is a (visually) trivial first of many non-trivial layout / design changes, so we may have to subclass `BrowserView` anyway. Though of course better to approach case-by-case.